pkg/integrations/v4/dm/emitter_no_register.go

```diff
@@ -1,124 +1,12 @@
-		metricsSender: dmSender,
-	}
-}
-
-func (e *nonRegisterEmitter) Send(dto fwrequest.FwRequest) {
-	entityRewrite := dto.EntityRewrite
-	integrationData := dto.Data
-
-	var emitErrs []error
-
-	plugin := agent.NewExternalPluginCommon(dto.PluginID(), e.agentContext, dto.Definition.Name)
-
-	labels, extraAnnotations := dto.LabelsAndExtraAnnotations()
-
-	var err error
-
-	emitV4DataSet := func(
-		idLookup host.IDLookup,
-		metricsSender MetricsSender,
-		emitter agent.PluginEmitter,
-		definition integration.Definition,
-		integrationMetadata protocol.IntegrationMetadata,
-		dataSet protocol.Dataset,
-		labels map[string]string,
-		extraAnnotations map[string]string,
-		entityRewrites data.EntityRewrites) error {
-
-		logEntry := elog.WithField("action", "EmitV4DataSet")
-
-		replaceEntityNameWithoutRegister := func(entity entity.Fields, entityRewrite data.EntityRewrites, idLookup host.IDLookup) error {
-
-			newName := entityRewrite.Apply(entity.Name)
-
-			agentShortName, err := idLookup.AgentShortEntityName()
-			newName = http.ReplaceLocalhost(newName, agentShortName)
-
-			if err != nil {
-				return err
-			}
-
-			entity.Name = newName
-			return nil
-		}
-
-		err := replaceEntityNameWithoutRegister(dataSet.Entity, entityRewrite, idLookup)
-		if err != nil {
-			return fmt.Errorf("error renaming entity: %s", err.Error())
-		}
-
-		integrationUser := definition.ExecutorConfig.User
-
-		if len(dataSet.Inventory) > 0 {
-			inventoryDataSet := legacy.BuildInventoryDataSet(
-				logEntry, dataSet.Inventory, labels, integrationUser, integrationMetadata.Name,
-				dataSet.Entity.Name)
-			emitter.EmitInventory(inventoryDataSet, entity.Entity{
-				Key: entity.Key(dataSet.Entity.Name),
-			})
-		}
-
-		for _, event := range dataSet.Events {
-			normalizedEvent := legacy.NormalizeEvent(elog, event, labels, extraAnnotations, integrationUser, dataSet.Entity.Name)
-
-			if normalizedEvent != nil {
-				emitter.EmitEvent(normalizedEvent, entity.Key(dataSet.Entity.Name))
-			}
-		}
-
-		dmProcessor := IntegrationProcessor{
-			IntegrationInterval:         definition.Interval,
-			IntegrationLabels:           labels,
-			IntegrationExtraAnnotations: extraAnnotations,
-		}
-		metricsSender.SendMetrics(dmProcessor.ProcessMetrics(dataSet.Metrics, dataSet.Common, dataSet.Entity))
-
-		return nil
-	}
-
-	for _, dataset := range integrationData.DataSets {
-		if err = emitV4DataSet(
-			e.agentContext.IDLookup(),
-			e.metricsSender,
-			&plugin,
-			dto.Definition,
-			integrationData.Integration,
-			dataset,
-			labels,
-			extraAnnotations,
-			entityRewrite,
-		); err != nil {
-			emitErrs = append(emitErrs, err)
-		}
-	}
-
-	// TODO error handling
-	composedError := composeEmitError(emitErrs, len(integrationData.DataSets))
-	if composedError != nil {
-		elog.Error(composedError.Error())
-	}
+func (e *emitter) processDatasetNoRegister(intMetadata protocol.IntegrationMetadata, reqMetadata fwrequest.FwRequestMeta, dataSet protocol.Dataset) {
```

Comment:
Now that there is only one emitter, does it make sense to have these functions separated in files?

Reply:
You are right, we could join them in the emitter.go file, but for now I would keep them separate as it might be more intuitive in case we need further implementations. Wdyt?